### ISSUE TITLE
Universal covering spaces, semi-locally simply connected, etc...

### ISSUE THREAD

**Issue by StevenClontz** [pi-base/data#818]
### Discussed in https://github.com/orgs/pi-base/discussions/817

<div type='discussions-op-text'>

<sup>Originally posted by **GeoffreySangston** October 21, 2024</sup>
(I apologize if this has already been discussed. I looked around and couldn't seem to see anything about it. Also, I've been using pi-base pretty consistently since college and think it's an incredibly helpful tool, so I just want to say thanks!)

I don't see the following properties. Is there a good reason for omitting these?

- [Contractible](https://en.wikipedia.org/wiki/Contractible_space)
- Locally contractible (on the same page as contractible)
- Strongly locally contractible (on the same page as contractible)
- [Simply connected](https://en.wikipedia.org/wiki/Simply_connected_space#Definition_and_equivalent_formulations)
- [Semi-locally simply connected](https://en.wikipedia.org/wiki/Semi-locally_simply_connected)
- Existence of a universal cover (Corollary 82.2 of Munkres says this is equivalent to path connected + locally path-connected + semi-locally simply connected)
- [Locally simply connected](https://en.wikipedia.org/wiki/Locally_simply_connected_space)
- [Acyclic](https://en.wikipedia.org/wiki/Acyclic_space)

(Edit: I think it's safe to say the standard definitions for the following depend on having a basepoint, since they depend on statements about the higher homotopy groups. So maybe that is a justification for not having them.)
- [Weakly contractible](https://en.wikipedia.org/wiki/Weakly_contractible)
- [Being an Eilenberg-MacLane space](https://en.wikipedia.org/wiki/Eilenberg%E2%80%93MacLane_space)
- [Aspherical](https://en.wikipedia.org/wiki/Aspherical_space)

And one could have properties of the fundamental group, in analogy to simply connected. For example, Hatcher has the following on page 49, "It is a theorem of [Shelah 1988] that for a path-connected, locally path-connected compact metric space X , π1(X) is either finitely generated or uncountable."

If these were added, pi-base should also have the [Hawaiian earring space](https://en.wikipedia.org/wiki/Hawaiian_earring). (There does appear to be a controversy around this name. Hatcher uses the descriptive name _The Shrinking Wedge of Circles_.)  And perhaps there should be the [rose with infinitely many petals](https://en.wikipedia.org/wiki/Rose_(topology)), which comes up when discussing the previous space. And I assume there's a whole canon of counterexamples here. I'd like to see spaces like the [pseudocircle](https://en.wikipedia.org/wiki/Pseudocircle), but perhaps that requires being able to make a more complex kind of query than pi-base desires to support (e.g., specifying the homotopy groups in some complex way).</div>

**Comment by GeoffreySangston** [comment]
I think that the "Has a universal cover" property may only really make sense for pointed spaces, in which case it would mean that the theorem constructing a simply connected cover using based paths applies. I think defining "a universal cover" as a simply connected cover (as Munkres and Hatcher actually do) isn't great because a simply connected space need not be locally path connected, as this [MSE page shows](https://math.stackexchange.com/questions/1526518/is-there-a-topological-space-which-is-simply-connected-but-not-local-path-connec).

**Comment by GeoffreySangston** [comment]
I'm experimenting with my own fork. I've added 'contractible' as a property and 'contractible => path connected' as a theorem.

I guess I see an issue from a pragmatic perspective. Pi-base currently has 59 path connected spaces. I'm not aware of any other property in pi-base which is implied by contractible and not already implied by path connected, though maybe the contributors here are. Is it okay upon first pull-request just to add the contractible property to a few of these (with sources), and let the others get added over time?

**Comment by prabau** [comment]
Yes, it's perfectly fine, and even recommended, to have a smaller PR that adds the definition with some theorems and maybe an example or two to illustrate.  And then add more in separate PRs.  A PR with a smaller scope makes it easier to review and easier to merge as well.

One general comment (Note: it's just my personal take on this).  It seems fine to me to add properties like contractible and others that can be expressed in a rather elementary way.  But properties that depend a lot more on higher order homotopy or homology groups, like acyclic, Eilenberg-MacLane space, aspherical, etc. require a lot more algebraic topology background and machinery.  Up to now, pi-base has been concentrating mostly on general topology, not algebraic topology.  Do we really want to have algebraic topology become part of pi-base?  Maybe, maybe not.  It's not for me to decide.  @StevenClontz should give direction on this.  Also, is there already a repository for algebraic topology equivalent to pi-base somewhere on the web?

**Comment by StevenClontz** [comment]
On principle, I have no problem with pi-Base modeling any property preserved by homeomorphism that's of interest to researchers. But we need community members who can confidently review contributions that rely on machinery from algebraic topology. I don't know if we're there yet.

https://mathbases.org/ is the best place to find other mathematical databases, and https://code4math.org is a place to connect with mathematicians from other disciplines interested in computing-enabled research.

**Comment by danflapjax** [comment]
It should be the case that a path-connected [LOTS](https://topology.pi-base.org/properties/P000133) is contractible iff it is separable (or [equivalently](https://topology.pi-base.org/spaces?q=LOTS%2BPath+connected%2BSeparable%2B%7EEmbeddable+in+%24%5Cmathbb+R%24) embeddable in $\mathbb R$). This can be seen from a [categorization of the possible such LOTS](https://math.stackexchange.com/a/4965665) and that the long ray isn't contractible (see [here](https://math.stackexchange.com/questions/1282097/prove-the-long-line-is-not-contractible) for example). I would imagine the forward and backward implications are generalizable, but I don't immediately see how.

**Comment by GeoffreySangston** [comment]
I would like to add simply connected with the following definition, as it seems to fit in with pi-base better by not involving concepts like fundamental groups and homotopy. I also don't refer to S000170 (Circle) or S000176 (Euclidean Plane R^2) since pi-base defines those up to homeomorphism and I don't see how to elegantly write this definition without referring to a specific circle / disk.

> $X$ is path connected, and every continuous map from the unit circle to $X$ is the restriction of a continuous map defined on the unit disk.

This is the first definition on Wikipedia and the second definition on [ncatlab](https://ncatlab.org/nlab/show/n-connected+space). It appears essentially in this form on page 51, Chapter 1.8 of Spanier's _Algebraic Topology_. Actually, Spanier writes
> A space $X$ is said to be _n-connected_ for $n \ge 0$ if every continuous map $f : S^k \to X$ for $k \le n$ has continuous extension over $E^{k+1}$. A $1$-connected space is also said to be _simply connected_.

I have a question though. The more common definition in the textbooks seems to be to involve the fundamental group (usually near the fact that fundamental groups at different base-points of path-connected spaces are isomorphic). This is done in Munkres's _Topology_, tom Dieck's _Algebraic Topology_, Massey's _A Basic Course in Algebraic Topology_, Rotman's _An Introduction to Algebraic Topology_. I don't see this defined in Engelking or Bourabaki's general topology books (though I'm not that familiar with these references). Also, _Encyclopedia of General Topology_ has a more general definition; it defines 'simply connected' to mean 'arcwise connected + trivial fundamental groups', and (I think) an arc there is a more general concept than in pi-bases P000038 (Arc connected).

 The text from Munkres is.
> A space $X$ is said to be **_simply connected_** if it is a path-connected space and if $\pi_1(X, x_0)$ is the trivial (one-element) group for some $x_0 \in X$, and hence for every $x_0 \in X$.


In addition to giving the first definition above (possibly with some stylistic revision), is it better to also include the fundamental group definition (from Munkres)? I don't see any results for 'fundamental group' when I use the search feature on the github.dev page.

**Comment by prabau** [comment]
(Side remark: FYI, the notion of "arc connected" will be see some revision in pi-base.  See the last comment in the discussion in #552.  The current "arc connected" will be renamed "injectively path connected", and the new "arc connected" will mean homeomorphically arc connected.  This will be a better match with the literature.)

**Comment by prabau** [comment]
pi-base did not mention the fundamental group anywhere before because it had nearly nothing involving algebraic topology.

Now if a notion is usually expressed in terms of the fundamental group (or even higher alg topol concepts), it would be fine and even recommended to phrase it in those terms.  At the same time, if there is also a way to express it in simpler terms, we can complement it by giving both formulations.  This can be backed up with references where the users can dig further as needed.

**Comment by felixpernegger** [comment]
@GeoffreySangston (and maybe @mathmaster13), we now have added most of the properties from the list. I was wondering if you know though, if "being an Eilenberg-MacLane space" or "acyclic" or "aspherical" (to a lesser degree also "weakly contractible") have nontrivial implications which can be added to pi-bade at all? (besides for CW complexes)
If not, adding them would probably leave tons of spaces unmarked, which is a bit unoptimal.

**Comment by GeoffreySangston** [comment]
@felixpernegger 

I'm guessing the impetus to adding these properties would come when many more manifolds are added to pi-base, if that ever happens. Eilenberg-MacLane space is probably premature. I think all of these properties are probably a while off from being added, but I'll add some comments.

I think homology properties like "acyclic" would have to be be split up for pi-base, because there are multiple homology theories to consider adding. E.g. the Warsaw circle has non-trivial first cech homology; see [MO](https://mathoverflow.net/q/165085/109654), but the Warsaw circle is weakly contractible, hence has trivial singular homology by the Hurewicz theorem.

There are a few theorems like property + weakly contractible => contractible. I.e., for property = cellular space (suggested multiple times now). This [MSE answer](https://math.stackexchange.com/a/4448131/444923) cites a paper by Milnor saying (metric) ANRs have the homotopy type of CW complexes, hence we can take property = ANR. I don't know if ANR has been suggested before, but maybe in the future it could get added. Also, Theorem 10.4 of Borsuk's _Theory of Retracts_ says "A finite-dimensional compactum $X$ is an ANR-space if and only if it is locally contractible". Compactum means compact metric space. I think it's likely we're adding in Borsuk's locally contractible sooner or later. The issue here is the lack of "Finite Lebesgue Covering Space Dimension", and dimension theory seems to be notoriously difficult to get added into pi-base because there are multiple versions to consider. The following would all be potential future theorems though:

- ANR* + weakly contractible* => contractible
-  Locally contractible* + compact + metrizable + Finite Lebesgue Covering Space Dimension* + weakly contractible => contractible* 
- Cellular space* + weakly contractible => contractible
(*'s mean the property doesn't exist in pi-base yet).

For aspherical we get the trivial one: "Aspherical + simply connected => weakly contractible". If any of the three theorems above makes it in eventually then this connected aspherical with contractible (in a trivial way though).

**Comment by prabau** [comment]
Is it fair to say that properties like acyclic, aspherical, Eilenberg-MacLane space are much more on the algebraic side of algebraic topology (vs. the topological side) and as such require a lot more algebraic topology machinery?  That would seem to veer away from the rest of pi-base.  Not that it would not be interesting eventually.

**Comment by felixpernegger** [comment]
I think this is fair to say. Notably all of these properties hold up to homotopy equivalence also.

**Comment by mathmaster13** [comment]
> Is it fair to say that properties like acyclic, aspherical, Eilenberg-MacLane space are much more on the algebraic side of algebraic topology (vs. the topological side) and as such require a lot more algebraic topology machinery? That would seem to veer away from the rest of pi-base. Not that it would not be interesting eventually.

I'd say so. The theory of Eilenberg-MacLane spaces is incredibly beautiful, but whether pi-base captures that beauty is its own question.

This being said, there are incredible algebraic counterexamples that do deserve a place to be mentioned, even if not pi base right now.

**Comment by felixpernegger** [comment]
@GeoffreySangston I wonder more generally what we ought to still be adding from this.
From the original list we still miss:

- Existence of a universal cover, see #1671
- Acyclic
- Weakly contractible, see #1761 (I just noticed that only 7 and 6 are switched lol)
- Eilenberg Maclane space
- Aspherical

Weakly contractible is i think highly useful due to Whitehead theorem. Similarly existence of universal cover is relatively easy to state. I also like (singular) acyclic, here we have weakly contractible => acyclic but also acyclic + simply connected => weakly contractible (modulo empty space potentially), both by Hurewicz theorem. (I doubt we have a space differentiating weakly contractible and acyclic right now). I'm not really sure if we really need different acyclic versions; its probably rather difficult to prove stuff about cech homology?

I do not think Aspherical and especially Eilenberg Maclane space admit some nontrivial theorems we can add. Similarly, I dont think there will be much on theorems about the fundamental group either (as has been noted in this thread before)

The best way to proceed in my opinion would be:

#1766 => Finite etc CW complexes => #1769 => #1761 => #1671 => acyclic

Once we have all that I think its fair to say implemented much of algebraic topology in pibase (which was definitely by far the biggest blindspot before in my opinion)

**Comment by felixpernegger** [comment]
Ive never used Cech homology, so I might be mistaken about some stuff FYI

**Comment by felixpernegger** [comment]
Another property which may be interesting would be "Homotopy equivalent to CW complex", which you kind of teased in [this](https://github.com/pi-base/data/issues/818#issuecomment-4025961888) comment. There seems to be [some](https://math.stackexchange.com/questions/1648250/spaces-homotopy-equivalent-to-finite-cw-complexes) (and [this](https://link.springer.com/article/10.1007/s11006-006-0031-9)) theory on this. If yes, this would generalise various theorems about CW complexes, since they often hold up to homotopy equivalence.

However I would only support this if there are indeed meaningful theorems for this.
